src/Struct/FindologicProduct.php

```diff
@@ -791,89 +909,27 @@ protected function getPricesFromProduct(ProductEntity $variant): array
             throw new ProductHasNoPricesException();
         }
 
         $this->prices = array_merge($this->prices, $prices);
     }
 
-    protected function setUrl(): void
-    {
-        $salesChannel = $this->salesChannelContext->getSalesChannel();
-
-        $domains = $salesChannel->getDomains();
-        $seoUrlCollection = $this->product->getSeoUrls()->filterBySalesChannelId($salesChannel->getId());
-        if ($domains && $domains->count() > 0 && $seoUrlCollection && $seoUrlCollection->count() > 0) {
-            $baseUrl = $domains->first()->getUrl();
-            $seoPath = $seoUrlCollection->first()->getSeoPathInfo();
-
-            $productUrl = sprintf('%s/%s', $baseUrl, $seoPath);
-        } else {
-            $productUrl = $this->router->generate(
-                'frontend.detail.page',
-                ['productId' => $this->product->getId()],
-                RouterInterface::ABSOLUTE_URL
-            );
-        }
-
-        $this->url = $productUrl;
-    }
-
-    protected function setKeywords(): void
-    {
-        $tags = $this->product->getTags();
-        if ($tags !== null && $tags->count() > 0) {
-            /** @var TagEntity $tag */
-            foreach ($tags as $tag) {
-                $this->keywords[] = new Keyword($tag->getName());
-            }
-        }
-    }
-
-    protected function setImages(): void
-    {
-        if (!$this->product->getMedia() || !$this->product->getMedia()->count()) {
-            $fallbackImage = $this->buildFallbackImage($this->router->getContext());
-
-            $this->images[] = new Image($fallbackImage);
-            $this->images[] = new Image($fallbackImage, Image::TYPE_THUMBNAIL);
-
-            return;
-        }
-
-        /** @var ProductMediaEntity $mediaEntity */
-        foreach ($this->getSortedImages() as $mediaEntity) {
-            if (!$mediaEntity->getMedia() || !$mediaEntity->getMedia()->getUrl()) {
-                continue;
-            }
-
-            $this->images[] = new Image($this->getEncodedUrl($mediaEntity->getMedia()->getUrl()));
-
-            $thumbnails = $mediaEntity->getMedia()->getThumbnails();
-            if (!$thumbnails) {
-                continue;
-            }
-
-            /** @var MediaThumbnailEntity $thumbnailEntity */
-            foreach ($thumbnails as $thumbnailEntity) {
-                $this->images[] = new Image($this->getEncodedUrl($thumbnailEntity->getUrl()), Image::TYPE_THUMBNAIL);
-            }
-        }
-    }
-
     /**
      * Takes invalid URLs that contain special characters such as umlauts, or special UTF-8 characters and
      * encodes them.
      */
     protected function getEncodedUrl(string $url): string
     {
         $parsedUrl = parse_url($url);
-
-        $parsedUrl['path'] = implode('/', array_map(
-            '\FINDOLOGIC\FinSearch\Utils\Utils::multiByteRawUrlEncode',
-            explode('/', $parsedUrl['path'])
-        ));
+        $parsedUrl['path'] = implode(
```

Comment:
Thats not well readable at all. Try to find a way to make this easier understandable.